src/pkg/packager/generate.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains functions for interacting with, managing and deploying Zarf packages.
+package packager
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// UpdatePackageConfigForGenerate updates the package configuration with the new ZarfPackageConfig from pkgConfig.GenerateOpts
+func UpdatePackageConfigForGenerate(pkgConfig *types.PackagerConfig) types.PackagerConfig {
+
+	// Add new ZarfPackageConfig
+	newComponent := []types.ZarfComponent{{
+		Name:     pkgConfig.GenerateOpts.Name,
+		Required: true,
+		Charts: []types.ZarfChart{
+			{
+				Name:      pkgConfig.GenerateOpts.Name,
+				Version:   pkgConfig.GenerateOpts.Version,
+				Namespace: pkgConfig.GenerateOpts.Name,
+				URL:       pkgConfig.GenerateOpts.URL,
+				GitPath:   pkgConfig.GenerateOpts.GitPath,
+			},
+		},
+	}}
+
+	pkgConfig.Pkg.Kind = types.ZarfPackageConfig
+	pkgConfig.Pkg.Metadata.Name = pkgConfig.GenerateOpts.Name
```

Comment:
```suggestion
	pkgConfig.Pkg.Metadata.Name = pkgConfig.GenerateOpts.Name
	pkgConfig.Pkg.Metadata.Version = pkgConfig.GenerateOpts.Version
```
It would likely be good to set the package version here too that way it is OCI-ready OOB.